pkg/runner/step_context.go

```diff
@@ -220,6 +221,58 @@ func (sc *StepContext) setupAction(actionDir string, actionPath string) common.E
 		if os.IsNotExist(err) {
 			f, err = os.Open(filepath.Join(actionDir, actionPath, "action.yaml"))
 			if err != nil {
+				if sc.Step.With != nil {
+					if val, ok := sc.Step.With["args"]; ok {
+						trampoline := []byte(`
+const { spawnSync } = require('child_process')
+const spawnArguments={
+  cwd: process.env['INPUT_CWD'],
+  stdio: [
+    process.stdin,
+    process.stdout,
+    process.stderr,
+  ]
+}
+const child=spawnSync(
+  '/bin/sh',
+  [ '-c' ].concat(process.env['INPUT_COMMAND']),
+  spawnArguments)
+process.exit(child.status)
+`)
+						err2 := ioutil.WriteFile(filepath.Join(actionDir, actionPath, "trampoline.js"), trampoline, 0755)
+						if err2 != nil {
+							return err
+						}
+						action := fmt.Sprintf(`
+name: '(Synthetic)'
+inputs:
+  cwd:
+    description: '(Actual working directory)'
+    default: >
+      %s
+  command:
+    description: '(Actual program)'
+    default: >
+      %s
+runs:
+  using: 'node12'
+  main: 'trampoline.js'
+`, filepath.Join(actionDir, actionPath), val)
+						actionYaml := filepath.Join(actionDir, actionPath, "action.yaml")
+						f, err2 = os.Create(actionYaml)
+						if err2 != nil {
+							return err
+						}
+						_, err2 = f.WriteString(action)
+						f.Close()
+						f, err2 = os.Open(actionYaml)
+						if err2 != nil {
+							return err
+						}
+						sc.Action, err = model.ReadAction(f)
```

Comment:
Seems unnecessary to write a file to then read it...why not just create the `Action` struct with the necessary values populated?

Reply:
My initial draft created an in memory string and passed that.

I'm not remotely comfortable w/ go. I'll have to look up how to properly fill in the struct by hand.